File: api/models/Ticket.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema

const TicketSchema = Schema({
    user_id: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    exhibition_id: { type: Schema.Types.ObjectId, ref: 'Exhibition', required: true },
    quantity: {type: Number, default: 1},
    status: { type: String, enum: ["ACTIVE", "COMPLETED", "CANCELLED"], default: "ACTIVE" },
    created_at: { type: Date, default: Date.now },
})

module.exports = mongoose.model("Ticket", TicketSchema)

File: app/lib/models/ticket.dart
class Ticket {
  final String? id;
  final String eventId;
  final int quantity;
  final String? status;

  Ticket({this.id, required this.eventId, required this.quantity, this.status});

  static Ticket fromJson(Map json) => Ticket(
      id: json["_id"],
      eventId: json["exhibition_id"],
      quantity: json["quantity"] as int,
      status: json["status"].toString());

  static Map toJson(Ticket ticket) =>
      {"exhibition_id": ticket.eventId, "quantity": ticket.quantity};
}
